lld/ELF/InputFiles.cpp

```diff
@@ -1418,6 +1434,36 @@ std::vector<uint32_t> SharedFile::parseVerneed(const ELFFile<ELFT> &obj,
   return verneeds;
 }
 
+// To determine if a shared file can support any of the GNU Attributes,
+// the .note.gnu.properties section need to be read. The appropriate
+// location in memory is located then the GnuPropertyNote can be parsed.
+// This is the same process as is used for readGnuProperty, however we
+// do not pass the data variable as, without an InputSection, its value
+// is unknown in a SharedFile. This is ok as the information that would
+// be collected from this is irrelevant for a dynamic object.
+template <typename ELFT>
+void SharedFile::parseGnuAndFeatures(const uint8_t *base,
+                                     const typename ELFT::PhdrRange headers) {
+  if (headers.size() == 0 || ctx.arg.emachine != EM_AARCH64)
+    return;
+
+  for (unsigned i = 0; i < headers.size(); i++) {
+    if (headers[i].p_type != PT_GNU_PROPERTY)
+      continue;
+    const typename ELFT::Note note(
+        *reinterpret_cast<const typename ELFT::Nhdr *>(base +
+                                                       headers[i].p_offset));
+    if (note.getType() != NT_GNU_PROPERTY_TYPE_0 || note.getName() != "GNU")
+      continue;
+
+    // Read a body of a NOTE record, which consists of type-length-value fields.
+    ArrayRef<uint8_t> desc = note.getDesc(headers[i].p_align);
+    parseGnuPropertyNote<ELFT>(
+        ctx, *this, /*featureAndType*/ GNU_PROPERTY_AARCH64_FEATURE_1_AND, desc,
```

Comment:
remove `/*featureAndType*/`. it's clear from the argument value (GNU_PROPERTY...)

If we need to name the argument, the canonical form is `/*xxx=*/value` (no space)